flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobMasterTest.java

```diff
@@ -1132,6 +1159,16 @@ private void runRequestNextInputSplitTest(
                             expectedRemainingInputSplits
                                     .apply(inputSplitsPerTask)
                                     .toArray(EMPTY_TESTING_INPUT_SPLITS));
+
+            // Make sure FailureEnrichers are triggered for the above failure
+            assertThat(testingEnricher.seenThrowable.stream().map(t -> t.getMessage()))
```

Comment:
Why is the failure enrichment tested in `testRequestNextInputSplit`? We should have a dedicated test for new features/regressions.